src/pybind/rados.py

```diff
@@ -1667,6 +1667,39 @@ def read(self, key, length=8192, offset=0):
             raise make_ex(ret, "Ioctx.read(%s): failed to read %s" % (self.name, key))
         return ctypes.string_at(ret_buf, ret)
 
+    @requires(('key', str), ('cls', str), ('method', str), ('data', str))
```

Comment:
with the recently-merged python 3 support, these types should distinguish between text and binary data.

seems like they should be:
key: str_type
cls: str_type
method: bytes
data: bytes

Then when calling rados_exec, use cstr(key) and cstr(cls)